hack/common.sh

```diff
@@ -498,12 +498,12 @@ os::build::ldflags() {
     os::build::get_version_vars
 
     declare -a ldflags=()
-    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.majorFromGit" "${OS_GIT_MAJOR}")
-    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.minorFromGit" "${OS_GIT_MINOR}")
-    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.versionFromGit" "${OS_GIT_VERSION}")
-    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.commitFromGit" "${OS_GIT_COMMIT}")
-    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitCommit" "${KUBE_GIT_COMMIT}")
-    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitVersion" "${KUBE_GIT_VERSION}")
+    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.majorFromGit=${OS_GIT_MAJOR}")
+    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.minorFromGit=${OS_GIT_MINOR}")
+    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.versionFromGit=${OS_GIT_VERSION}")
+    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.commitFromGit=${OS_GIT_COMMIT}")
+    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitCommit=${KUBE_GIT_COMMIT}")
+    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitVersion=${KUBE_GIT_VERSION}")
```

Comment:
It's not compatible between 1.4 and 1.5.  You'll have to have a conditional
check in the build and use the old format for < 1.5

On Oct 2, 2015, at 11:32 AM, Ben Parees <notifications@github.com> wrote:

In hack/common.sh
<https://github.com/openshift/origin/pull/4906#discussion_r41034332>:

> @@ -498,12 +498,12 @@ os::build::ldflags() {
>      os::build::get_version_vars
>
>      declare -a ldflags=()
> -    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.majorFromGit" "${OS_GIT_MAJOR}")
> -    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.minorFromGit" "${OS_GIT_MINOR}")
> -    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.versionFromGit" "${OS_GIT_VERSION}")
> -    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.commitFromGit" "${OS_GIT_COMMIT}")
> -    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitCommit" "${KUBE_GIT_COMMIT}")
> -    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitVersion" "${KUBE_GIT_VERSION}")
> +    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.majorFromGit=${OS_GIT_MAJOR}")
> +    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.minorFromGit=${OS_GIT_MINOR}")
> +    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.versionFromGit=${OS_GIT_VERSION}")
> +    ldflags+=(-X "${OS_GO_PACKAGE}/pkg/version.commitFromGit=${OS_GIT_COMMIT}")
> +    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitCommit=${KUBE_GIT_COMMIT}")
> +    ldflags+=(-X "k8s.io/kubernetes/pkg/version.gitVersion=${KUBE_GIT_VERSION}")

was this explicilty needed for 1.5?

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/4906/files#r41034332>.